### ISSUE TITLE
Setting `JAX_PLATFORMS=cpu` has no effect.

### ISSUE THREAD

**Issue by g199209** [google/jax#23208]
### Description

I have installed `jax[cuda12]`, but not all hosts have GPU, so I want to use CPU on there machines.

I tried to use this way first:

```python
def jax_force_use_cpu_v1():
    os.environ["JAX_PLATFORMS"] = "cpu"
    jax.config.update("jax_platform_name", 'cpu')

jax_force_use_cpu_v1()
```

But this hasn't work:

```
RuntimeError: Unable to initialize backend 'cuda': FAILED_PRECONDITION: No visible GPU devices. (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)
```

I found another workaround later:

```python
def jax_force_use_cpu_v2():
    try:
        jax.lib.xla_bridge.get_backend()
    except RuntimeError:
        pass

jax_force_use_cpu_v2()
```

This works but it looks weird. I'm not sure is this the correct way to do it?

### System info (python version, jaxlib version, accelerator, etc.)

jax varsion: 0.4.30

**Comment by yashk2810** [comment]
JAX_PLATFORMS=cpu is an env var. You need to set it before you import jax via `os.environ`.

**Comment by g199209** [comment]
> JAX_PLATFORMS=cpu is an env var. You need to set it before you import jax via `os.environ`.

This works, thanks. But what if I want to use the CPU only when the GPU is unavailable?

**Comment by yashk2810** [comment]
Then you can do: `JAX_PLATFORMS=gpu,cpu`

**Comment by yashk2810** [comment]
Also, I think if you upgrade to 0.4.31, then you don't need this hack. We fixed it in 0.4.31 IIRC. It should automatically fallback to CPU if GPU is not present.

**Comment by yashk2810** [comment]
I am closing the issue but feel free to reopen if it doesn't work!

**Comment by g199209** [comment]
> Also, I think if you upgrade to 0.4.31, then you don't need this hack. We fixed it in 0.4.31 IIRC. It should automatically fallback to CPU if GPU is not present.

Nice to hear that. I'll try 0.4.31 later.
